fou_test.go

```diff
@@ -0,0 +1,103 @@
+package netlink
+
+import (
+	"testing"
+)
+
+func TestFouDeserializeMsg(t *testing.T) {
+	var msg []byte
+
+	// deserialize a valid message
+	msg = []byte{3, 1, 0, 0, 5, 0, 2, 0, 2, 0, 0, 0, 6, 0, 1, 0, 21, 179, 0, 0, 5, 0, 3, 0, 4, 0, 0, 0, 5, 0, 4, 0, 1, 0, 0, 0}
+	if fou, err := deserializeFouMsg(msg); err != nil {
+		t.Error(err.Error())
+	} else {
+
+		// check if message was deserialized correctly
+		if fou.Family != FAMILY_V4 {
+			t.Errorf("expected family %d, got %d", FAMILY_V4, fou.Family)
+		}
+
+		if fou.Port != 5555 {
+			t.Errorf("expected port 5555, got %d", fou.Port)
+		}
+
+		if fou.Protocol != 4 { // ipip
+			t.Errorf("expected protocol 4, got %d", fou.Protocol)
+		}
+
+		if fou.EncapType != FOU_ENCAP_DIRECT {
+			t.Errorf("expected encap type %d, got %d", FOU_ENCAP_DIRECT, fou.EncapType)
+		}
+	}
+
+	// deserialize truncated attribute header
+	msg = []byte{3, 1, 0, 0, 5, 0}
+	if _, err := deserializeFouMsg(msg); err == nil {
+		t.Error("expected attribute header truncated error")
+	} else if err != ErrAttrHeaderTruncated {
+		t.Errorf("unexpected error: %s", err.Error())
+	}
+
+	// deserialize truncated attribute header
+	msg = []byte{3, 1, 0, 0, 5, 0, 2, 0, 2, 0, 0}
+	if _, err := deserializeFouMsg(msg); err == nil {
+		t.Error("expected attribute body truncated error")
+	} else if err != ErrAttrBodyTruncated {
+		t.Errorf("unexpected error: %s", err.Error())
+	}
+}
+
+func TestFouAddDel(t *testing.T) {
+	tearDown := setUpNetlinkTest(t)
```

Comment:
Please add the call to `minKernelRequired(t, M, N)` so that this test only runs on kernel which knows about FOU